graphql-dgs-spring-webflux-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/webflux/autoconfiguration/DgsWebFluxAutoConfiguration.kt

```diff
@@ -0,0 +1,194 @@
+        val mutationExecutionStrategy =
+            providedMutationExecutionStrategy.orElse(AsyncSerialExecutionStrategy(dataFetcherExceptionHandler))
+        return DefaultDgsReactiveQueryExecutor(
+            schema,
+            schemaProvider,
+            dgsDataLoaderProvider,
+            dgsContextBuilder,
+            chainedInstrumentation,
+            queryExecutionStrategy,
+            mutationExecutionStrategy,
+            idProvider,
+            reloadSchemaIndicator
+        )
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    open fun reactiveGraphQlContextBuilder(
+        dgsReactiveCustomContextBuilderWithRequest: Optional<DgsReactiveCustomContextBuilderWithRequest<*>>
+    ): DefaultDgsReactiveGraphQLContextBuilder {
+        return DefaultDgsReactiveGraphQLContextBuilder(dgsReactiveCustomContextBuilderWithRequest)
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = ["dgs.graphql.graphiql.enabled"], havingValue = "true", matchIfMissing = true)
+    open fun graphiQlConfigurer(configProps: DgsWebfluxConfigurationProperties): GraphiQlConfigurer {
+        return GraphiQlConfigurer(configProps)
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = ["dgs.graphql.graphiql.enabled"], havingValue = "true", matchIfMissing = true)
+    open fun graphiQlIndexRedirect(): RouterFunction<ServerResponse> {
+        return RouterFunctions.route().GET("/graphiql") {
+            permanentRedirect(URI.create("/graphiql/index.html")).build()
+        }.build()
+    }
+
+    @Bean
+    open fun dgsGraphQlRouter(dgsQueryExecutor: DgsReactiveQueryExecutor): RouterFunction<ServerResponse> {
+        val graphQlHandler = GraphQlHandler(dgsQueryExecutor)
+
+        return RouterFunctions.route()
+            .POST(
+                configProps.path, accept(MediaType.APPLICATION_JSON, MediaType.valueOf("application/graphql")),
+                graphQlHandler::graphql
+            ).build()
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = ["dgs.graphql.schema-json.enabled"], havingValue = "true", matchIfMissing = true)
+    open fun schemaRouter(schemaProvider: DgsSchemaProvider): RouterFunction<ServerResponse> {
+        return RouterFunctions.route()
+            .GET(
+                configProps.schemaJson.path
+            ) {
+                val graphQLSchema: GraphQLSchema = schemaProvider.schema()
+                val graphQL = GraphQL.newGraphQL(graphQLSchema).build()
+
+                val executionInput: ExecutionInput =
+                    ExecutionInput.newExecutionInput().query(IntrospectionQuery.INTROSPECTION_QUERY)
+                        .build()
+                val execute = graphQL.executeAsync(executionInput)
+
+                return@GET Mono.fromCompletionStage(execute)
+                    .map { it.toSpecification() }
+                    .flatMap { ok().json().bodyValue(it) }
+            }.build()
+    }
+
+    class GraphQlHandler(private val dgsQueryExecutor: DgsReactiveQueryExecutor) {
```

Comment:
Don't you think about move it to another class an make it open? Then inject in here.
I suppose, other people will be glad to add additional logic in it)

Reply:
Moved to another class for clarity. Leaving injection (as a hook to override) for later, as there are some decisions to make related to it.